__tests__/app.test.js

```diff
@@ -312,3 +313,125 @@ describe('POST /api/articles/:article_id/comments', () => {
+        return request(app)
+            .patch(`/api/articles/${articleId}`)
+            .send(patch)
+            .expect(201)
+            .then(({body}) => {
+                expect(typeof body.article).toBe('object');
+                expect(Array.isArray(body.article)).not.toBe(true);
+                expect(body.article).toMatchObject(expectedOutput);
+                expect(body.article.votes).toBe(expectedOutput.votes);
+            });
+    });
+    test('201: should return an unchanged article when votes not changed', () => {
+        const articleId = 1;
+        const patch = {inc_votes: 0};
+        const expectedOutput = {...data.articleData[articleId - 1]};
+        expectedOutput.votes += patch.inc_votes;
+        // adjust for hour's difference between dataset and database
+        const createDate = new Date(expectedOutput.created_at);
+        expectedOutput.created_at = new Date(createDate.setHours(createDate.getHours() - 1)).toISOString();
+        return request(app)
+            .patch(`/api/articles/${articleId}`)
+            .send(patch)
+            .expect(201)
+            .then(({body}) => {
+                expect(typeof body.article).toBe('object');
+                expect(Array.isArray(body.article)).not.toBe(true);
+                expect(body.article).toMatchObject(expectedOutput);
+                expect(body.article.votes).toBe(expectedOutput.votes);
+            });
+    });
+    test('201: should return an updated patched article with an increase in votes if supplied as a string', () => {
+        const articleId = 1;
+        const patch = {inc_votes: "100"};
+        const expectedOutput = {...data.articleData[articleId - 1]};
+        expectedOutput.votes += Number(patch.inc_votes);
+        // adjust for hour's difference between dataset and database
+        const createDate = new Date(expectedOutput.created_at);
+        expectedOutput.created_at = new Date(createDate.setHours(createDate.getHours() - 1)).toISOString();
+        return request(app)
+            .patch(`/api/articles/${articleId}`)
+            .send(patch)
+            .expect(201)
+            .then(({body}) => {
+                expect(typeof body.article).toBe('object');
+                expect(Array.isArray(body.article)).not.toBe(true);
+                expect(body.article).toMatchObject(expectedOutput);
+                expect(body.article.votes).toBe(expectedOutput.votes);
+            });
+    });
+    test('400: should return bad request error if non-numeric votes provided', () => {
+        const articleId = 1;
+        const patch = {inc_votes: "invalid input"};
+        return request(app)
+            .patch(`/api/articles/${articleId}`)
+            .send(patch)
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe('Bad Request');
+            });
+    });
+    test('400: should return bad request error if no inc_votes key provided', () => {
```

Comment:
Status 200, if inc_votes is missing, should not update article and return original single article object (this may be included as a 400 but that could make for a brittle test suite)